tools/fast/fastesting/basic.go

```diff
@@ -77,8 +77,18 @@ func NewTestEnvironment(ctx context.Context, t *testing.T, fastenvOpts fast.File
 	// Define a MiningOnce function which will bet set on the context to provide
 	// a way to mine blocks in the series used during testing
 	var MiningOnce series.MiningOnceFunc = func() {
-		_, err := genesis.MiningOnce(ctx)
+		isActive, err := genesis.MiningStatus(ctx)
 		require.NoError(t, err)
+		if isActive {
+			err = genesis.MiningStop(ctx)
```

Comment:
Blocking: this might be the right way but I am suspicious this will cause confusion.  It looks like this is not violating any existing test assumptions which is a good sign.  The alternative would be to make this fail when called during active mining and handle the complexity a level up.

cc @travisperson could you either sign off or suggest changes here?  @frrist you can make this call too if you feel high context.

Reply:
I am a fan of handling this a level up as it feels like this could lead to some odd behaviour, but I will defer to @travisperson.

Reply:
I think the best thing to do here would be to have the user set their own mining once function when they are going to explicitly use the async mining by calling `ctx = series.SetCtxMiningOnce(ctx, func() {})` after they setup the environment, as the preferred method of using this package is through mining once, and the use of `mining start` anywhere at the moment is an issue that will be address in #2968.

So for now, we should update the other tests, and not change this.